mongomock/aggregate.py

```diff
@@ -178,46 +181,84 @@ def _parse_basic_expression(self, expression):
+            try:
+                return None if self.parse(values) is None else unary_operators[operator](self.parse(values))
+            except KeyError:
+                return None
+        if operator in binary_operators:
+            if not isinstance(values, list):
+                raise OperationFailure('Expression %s takes exactly 2 arguments. 1 were passed in.' % operator)
+            if len(values) != 2:
+                raise OperationFailure('Expression %s takes exactly 2 arguments. %d were passed in.'
+                                       % (operator, len(values)))
+            try:
+                values = list(map(self.parse, values))
+            except KeyError:
+                return None
+            if None in values:
+                return None
+            if operator == '$subtract' and isinstance(values[0], datetime.datetime):
+                if isinstance(values[1], datetime.datetime):
+                    return round((values[0] - values[1]).total_seconds() * 1000)
+                return values[0] - datetime.timedelta(milliseconds=values[1])
+            return binary_operators[operator](values[0], values[1])
         if operator == '$add':
-            return sum(self.parse(value) for value in values)
-        if operator == '$ceil':
-            return math.ceil(self.parse(values))
-        if operator == '$divide':
-            assert len(values) == 2, 'divide must have only 2 items'
-            return self.parse(values[0]) / self.parse(values[1])
-        if operator == '$exp':
-            return math.exp(self.parse(values))
-        if operator == '$floor':
-            return math.floor(self.parse(values))
-        if operator == '$ln':
-            return math.log(self.parse(values))
-        if operator == '$log':
-            assert len(values) == 2, 'log must have only 2 items'
-            return math.log(self.parse(values[0]), self.parse(values[1]))
-        if operator == '$log10':
-            return math.log10(self.parse(values))
-        if operator == '$mod':
-            assert len(values) == 2, 'mod must have only 2 items'
-            return math.fmod(self.parse(values[0]), self.parse(values[1]))
+            if not isinstance(values, list):
+                values = [values]
+            parsed_dates = []
+            parsed_numbers = []
+            for value in values:
+                try:
+                    value = self.parse(value)
+                    if value is None:
+                        return None
+                    elif isinstance(value, datetime.datetime):
+                        parsed_dates.append(value)
+                    else:
+                        parsed_numbers.append(value)
+                except KeyError:
+                    return None
+            if len(parsed_dates) > 1:
+                raise OperationFailure('only one date allowed in an $add expression')
+            result = sum(parsed_numbers)
+            if len(parsed_dates) == 1:
```

Comment:
`if parsed_dates:` should be enough of a check as you already handled the case where there were more than 1.

Reply:
Dropped.